File: engine/core/base/events.cpp
#include "events.hpp"

namespace r2d2 {

void EventQueue::push(const Event& e) {
    queue_.push(e);
}

bool EventQueue::poll(Event& out) {
    if (queue_.empty())
        return false;

    out = queue_.front();
    queue_.pop();
    return true;
}

void EventQueue::clear() {
    while (!queue_.empty())
        queue_.pop();
}

} // namespace r2d2

File: engine/python/base/events.py
from engine.cython.base.events import Events as _Events

class Events:
    def __init__(self):
        self._ev = _Events()

    def push(self, type: int, a: int = 0, b: int = 0):
        self._ev.push(type, a, b)

    def poll(self):
        return self._ev.poll()

    def clear(self):
        self._ev.clear()
